Library/Homebrew/rubocops/cask/constants/stanza.rb

```diff
@@ -6,7 +6,7 @@ module Cask
     # Constants available globally for use in all cask cops.
     module Constants
       STANZA_GROUPS = [
-        [:arch, :on_arch_conditional],
+        [:on_arm, :on_intel, :arch, :on_arch_conditional],
```

Comment:
Is this too restrictive? I noticed there's more `on_arch` and `on_system` methods for eg. previous macOS versions? Should I make this more general using all the values of, say, https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/cask/constants/stanza.rb#L59? But then what about nesting block stanza order (which the stanza order cop [doesn't currently support](https://github.com/Homebrew/brew/issues/14017))?

Reply:
Yes, I think so. Take a look at [`orion`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/orion.rb) to see how we have to do ths `on_system` sometimes.

Reply:
Wow, that's a lot!

Reply:
There are a couple of examples where this could get tripped up, because of the way the `on_system` blocks work...

Teamviewer is an absolute mess of a cask, but it's an example of where we have moved the order of the stanzas for legibility.
https://github.com/homebrew/homebrew-cask/blob/a4ad99eb97ae00656893ab07e1313244f1ffca5f/Casks/teamviewer.rb